src/lib/api.js

```diff
@@ -1,77 +1,114 @@
+        error.response = response; // Need to pass response through so we can see what kind of HTTP error this was
+        throw error;
+    }
+};
+
+/**
+ * Pass through transformer if the response type is text
+ * @param {Object} data
+ * @return {Object}
+ */
+const transformTextResponse = (data) => data;
+
+class Api {
+    responseInterceptorId;
+    requestInterceptorId;
 
     /**
-     * Helper function to convert an Axios error to the format Preview expects
+     * [constructor]
      *
-     * @private
-     * @param {Object} response - Axios error response
-     * @throws {Error} - Throws when an error response object exists
-     * @return {void}
+     * @return {Api} Instance of the API
      */
-    handleError({ response }) {
-        if (response) {
-            const error = new Error(response.statusText);
-            error.response = response; // Need to pass response through so we can see what kind of HTTP error this was
-            throw error;
-        }
-    },
+    constructor() {
+        this.client = axios.create();
+    }
 
     /**
-     * Retrieves JSON from response.
-     *
-     * @private
-     * @param {Response} response - Response to parse
-     * @return {Promise|Response} Response if 204 or 202, otherwise promise that resolves with JSON
+     * Adds a function that intercepts an http response
+
+     * @public
+     * @param {Function} responseInterceptor - Function that gets called on each response
+     * @return {void}
      */
-    parseResponse: (response) => {
-        if (response.status === 204 || response.status === 202) {
-            return response;
+    addResponseInterceptor(responseInterceptor) {
+        if (typeof responseInterceptor === 'function') {
+            this.responseInterceptorId = this.client.interceptors.response.use(responseInterceptor);
         }
+    }
 
-        return response.data;
-    },
+    /**
+     * Adds a function that intercepts an http request
+     * @public
+     * @param {Function} requestInterceptor - function that gets called on each request
+     * @return {void}
 
-    transformTextResponse: (data) => data,
+     */
+    addRequestInterceptor(requestInterceptor) {
+        if (typeof requestInterceptor === 'function') {
+            this.requestInterceptorId = this.client.interceptors.request.use(requestInterceptor);
+        }
+    }
 
     /**
-     * Wrapper function for XHR post put and delete
+     * Ejects all interceptors
+     * @public
      *
-     * @private
-     * @param {string} url - The URL for XHR
-     * @param {Object} options - The request options
-     * @return {Promise} - XHR promise
+     * @return {void}
      */
-    xhr(url, options = {}) {
-        let transformResponse;
-
-        if (options.responseType === 'text') {
-            transformResponse = api.transformTextResponse;
-        }
+    ejectInterceptors() {
+        this.client.interceptors.response.eject(this.responseInterceptorId);
```

Comment:
Axios supports multiple interceptors and this will track only the last one added. We discussed previously that we'd iterate over the interceptors and remove them all.